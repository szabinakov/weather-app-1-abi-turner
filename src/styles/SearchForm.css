 
  button {
    flex: 1 0 50px;
    max-height: 2em;
    font-size: 0.7em;
    letter-spacing: .2em;
    text-transform: uppercase;
    border-radius: 0px 5px 5px 0px;
      background: silver;
    color: white;
    border: none;
    padding: 0.35em 0.75em;
    cursor: pointer;
  
  }
  button:hover {
    background: skyblue;
  }

  .search-form {
    
    -ms-flex-align: center;
    
  }

  /* .search-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
  } */
  
  .search-form {
    width: 300px;
    height: 40px;
    border: none;
    border-radius: 20px;
    padding: 10px;
    font-size: 16px;
  }
  