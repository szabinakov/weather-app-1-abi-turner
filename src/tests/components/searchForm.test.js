import React from "react";
import { render, screen } from "@testing-library/react";
import SearchForm from "../../components/searchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Leeds",
    setSearchText: () => {},
    onSubmit: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("button displays the correct text", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.searchText}
        onSubmit={validProps.onSubmit}
      />
    );
    const button = screen.getByRole("button", { name: /search/i });

    expect(button).toBeInTheDocument();
  });
});
