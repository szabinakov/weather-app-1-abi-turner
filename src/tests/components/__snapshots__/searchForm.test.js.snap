// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value="Leeds"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
